### PR TITLE
fix ODR violation of global LLVMZeroValueCTX and LLVMZeroValueMod

### PR DESCRIPTION
**Pull request by MMory**

### CODE REVIEW

File: include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/LLVMZeroValue.h [hunk 1]
```diff
@@ -33,9 +33,9 @@ class Value;
 namespace psr {
 
 // do not touch, its only purpose is to make ZeroValue working
-static const std::unique_ptr<llvm::LLVMContext>
+inline const std::unique_ptr<llvm::LLVMContext>
     LLVMZeroValueCTX(new llvm::LLVMContext);
-static const std::unique_ptr<llvm::Module>
+inline const std::unique_ptr<llvm::Module>
```
Comment (vulder) [inline include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/LLVMZeroValue.h:38]:
```suggestion
static inline const std::unique_ptr<llvm::Module>
```

**Review by vulder — CHANGES_REQUESTED** [review]

File: include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/LLVMZeroValue.h [hunk 2]
```diff
@@ -33,9 +33,9 @@ class Value;
 namespace psr {
 
 // do not touch, its only purpose is to make ZeroValue working
-static const std::unique_ptr<llvm::LLVMContext>
+inline const std::unique_ptr<llvm::LLVMContext>
```
[the remark this answers is not in the document]
Comment (MMory) [reply include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/LLVMZeroValue.h:36]: nope. by doing so we have one instance per compiled cpp file, which is wrong.

[the remark this answers is not in the document]
Comment (vulder) [reply include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/LLVMZeroValue.h:36]: but with the inline all these instances are collapsed during linking, so the final binary only has one instance

[the remark this answers is not in the document]
Comment (MMory) [reply include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/LLVMZeroValue.h:36]: they are not. This is what I expected too, but in the debugger I could see there is one instance per compiled file. The collapsing takes place for members iirc, but those are no members.

[the remark this answers is not in the document]
Comment (vulder) [reply include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/LLVMZeroValue.h:36]: ah, yeah the inline static only is relevant for static members variables, for globals the inline is enough

**Review by vulder — APPROVED** [review]

**Review by pdschubert — APPROVED** [review]

**Pull request merged**